assets/lib/browser-support.js

```diff
@@ -8,34 +8,59 @@
             valid = undefined;
         }
 
+        if (valid === undefined && console) console.log('Missing JS feature: ' + name);
         return (valid !== undefined);
     }
 
     function css(prop, val) {
         var valid;
 
         try {
-            valid = CSS.supports(prop, val);
+            if (val) {
+                valid = CSS && CSS.supports && CSS.supports(prop, val);
+            } else {
+                // Support query string (instead of prop:val)
+                valid = CSS && CSS.supports && CSS.supports(prop);
+            }
         } catch (err) {
             valid = false;
         }
 
+        if (!valid && console) console.log('Missing CSS feature: ' + [prop, val].join(', '));
         return valid;
     }
 
     function browserCheck() {
         var browserTest = true,
             valid;
 
-        {% for rule in site.data.browser_support.rules %}{% if rule[1].req %}
-        /* Check {{ rule[0] }}: {{ rule[1].req[0] }} in {{ rule[1].req[1] }} */
-        if (!req('{{ rule[1].req[0] }}', {{ rule[1].req[1] }})) browserTest = false;
-        {% endif %}{% if rule[1].css %}
-        /* Check {{ rule[0] }}: @supports({{ rule[1].css[0] }}:{{ rule[1].css[1] }}) */
-        if (!css('{{ rule[1].css[0] }}', '{{ rule[1].css[1] }}')) browserTest = false;
-        {% endif %}{% endfor %}
+        return ({% for rule in site.data.browser_support.rules %}{%
+            if rule[1].req 
 
-        return browserTest;
+                %}
+            /* Check {{ rule[0] }}: {{ rule[1].req[0] }} in {{ rule[1].req[1] }} */
+            req('{{ rule[1].req[0] }}', {{ rule[1].req[1] }}) && {%
+
+            elsif rule[1].css
+            %}{%
+                if rule[1].css[1] 
+
+                    %}
+            /* Check {{ rule[0] }}: @supports({{ rule[1].css[0] }}:{{ rule[1].css[1] }}) */
+            css('{{ rule[1].css[0] }}', '{{ rule[1].css[1] }}') && {%
+
+                else
+
+                    %}
+            /* Check {{ rule[0] }}: @supports({{ rule[1].css }}) */
+            css('{{ rule[1].css }}') && {%
+
+                endif
+                %}{%
+            endif
+            %}{%
+        endfor %}
+        true);
```

Comment:
Oh, and I have a stray true at the end, as I output && on each check, rather than doing some weird assignment dance between Liquid and Javascript and doing a join at the end. So it does check for "true" at the end, but that's expected. :rofl:

Reply:
I don't follow, I'm afraid -- does this need to be fixed, or is good as-is?

Reply:
No, this is fine. I'm just pointing out that it's fine even though it looks really funky.

It's Liquid templating pulling data from Ruby-generated YAML to generate JS to run in the browser. But: It works and it's the same rules we're using in the login page.

Reply:
Ack, thank you!